### ISSUE TITLE
Safari/VoiceOver reads out headings within legends twice

### ISSUE THREAD

**Issue by lfdebrux** [alphagov/govuk-frontend#2395]
## Description of the issue

When using VoiceOver with Safari on iOS or macOS headings that are contained within legends will be read out twice.

For instance, with iOS, on the [question page example](https://design-system.service.gov.uk/patterns/question-pages/default/index.html), VoiceOver announces

> Where do you live? Heading level one. Where do you live? Landmark. Form start.

May be related to https://github.com/alphagov/govuk-frontend/issues/2153, may be fixed by https://github.com/alphagov/govuk-frontend/issues/1605. Raised in question page backlog discussion  https://github.com/alphagov/govuk-design-system-backlog/issues/58#issuecomment-945603738.

## Steps to reproduce the issue

1. On macOS, enable VoiceOver and open Safari
2. Visit https://design-system.service.gov.uk/patterns/question-pages/default/index.html
3. Move VoiceOver to the top of the page
4. Start VoiceOver reading with keyboard shortcut Control + Option + A

## Actual vs expected behaviour

Actual behaviour:

When VoiceOver reaches the page heading, it announces
> Where do you live?, group heading level 1 Where do you live?

Expected behaviour:

VoiceOver announces the page heading 'Where do you live?' only once

## Environment (where applicable)

- Operating system: macOS Big Sur Version 11.6
- Browser: Safari
- Browser version: Version 15.0 (16612.1.29.41.4, 16612)
- GOV.UK Frontend Version: 3.14.0

I tried similar steps with NVDA with Firefox on Windows and JAWS with Chrome on Windows, and was unable to reproduce the issue. This issue may be a bug in VoiceOver/Safari.

**Comment by querkmachine** [comment]
Checking in to acknowledge that as of macOS Tahoe (26.2) and Safari 26.2, this issue is still present. 

This seems to be because the legend is announced first when the `<fieldset>` element receives focus, and again when the `<h1>` element subsequently receives focus. It does not duplicate the announcement if either one of these is skipped (e.g. going directly to the `<h1>` via the rotor).
